app/uk/gov/hmrc/trackingconsentfrontend/assets/test/interfaces/optimizelyCommunicatorFactory.spec.ts

```diff
@@ -68,4 +106,85 @@ describe('sendPreferences', () => {
       { type: 'optOut', isOptOut: true }
     ])
   })
+
+  it('should setup optimizely when configured', () => {
+    const control = setupScriptWithID('12345678901', exampleTCScriptURL)
+    control.trigger()
+
+    expect(getScriptSrcs())
+      .toEqual([
+        exampleTCScriptURL,
+        'https://cdn.optimizely.com/js/12345678901.js'
+      ])
+    expect(console.error).not.toHaveBeenCalled()
+
+    control.cleanup()
+  })
+
+  it('should setup optimizely when configured', () => {
+    const control = setupScriptWithID('98765432109', exampleTCScriptURL)
+    control.trigger()
+
+    expect(getScriptSrcs())
+      .toEqual([
+        exampleTCScriptURL,
+        'https://cdn.optimizely.com/js/98765432109.js'
+      ])
+    expect(console.error).not.toHaveBeenCalled()
+
+    control.cleanup()
+  })
+
+  it('should not setup optimizely when configured on the wrong script', () => {
```

Comment:
I would question whether we need to support this case and introduces the need to have that complicated regex or something similar. For me, it's sufficient to simply filter the script tags on data-optimizely-id.